src/modules/silentpayments/main_impl.h

```diff
@@ -7,10 +7,309 @@
+    } else {
+        ARG_CHECK(n_plain_seckeys == 0);
+    }
+    for (i = 0; i < n_recipients; i++) {
+        ARG_CHECK(recipients[i]->index == i);
+    }
+
+    seckey_sum_scalar = secp256k1_scalar_zero;
+    for (i = 0; i < n_plain_seckeys; i++) {
+        ret = secp256k1_scalar_set_b32_seckey(&addend, plain_seckeys[i]);
+        secp256k1_declassify(ctx, &ret, sizeof(ret));
+        if (!ret) {
+            secp256k1_scalar_clear(&addend);
+            secp256k1_scalar_clear(&seckey_sum_scalar);
+            return 0;
+        }
+        secp256k1_scalar_add(&seckey_sum_scalar, &seckey_sum_scalar, &addend);
+    }
+    /* Private keys used for taproot outputs have to be negated if they result in an odd point. This is to ensure
+     * the sender and recipient can arrive at the same shared secret when using x-only public keys. */
+    for (i = 0; i < n_taproot_seckeys; i++) {
+        secp256k1_ge addend_point;
+        ret = secp256k1_keypair_load(ctx, &addend, &addend_point, taproot_seckeys[i]);
+        secp256k1_declassify(ctx, &ret, sizeof(ret));
+        if (!ret) {
+            secp256k1_scalar_clear(&addend);
+            secp256k1_scalar_clear(&seckey_sum_scalar);
+            return 0;
+        }
+        if (secp256k1_fe_is_odd(&addend_point.y)) {
+            secp256k1_scalar_negate(&addend, &addend);
+        }
+        secp256k1_scalar_add(&seckey_sum_scalar, &seckey_sum_scalar, &addend);
+    }
+    /* If there are any failures in loading/summing up the secret keys, fail early. */
+    sum_is_zero = secp256k1_scalar_is_zero(&seckey_sum_scalar);
+    secp256k1_declassify(ctx, &sum_is_zero, sizeof(sum_is_zero));
+    secp256k1_scalar_clear(&addend);
+    if (sum_is_zero) {
+        secp256k1_scalar_clear(&seckey_sum_scalar);
+        return 0;
+    }
+    secp256k1_ecmult_gen(&ctx->ecmult_gen_ctx, &prevouts_pubkey_sum_gej, &seckey_sum_scalar);
+    secp256k1_ge_set_gej(&prevouts_pubkey_sum_ge, &prevouts_pubkey_sum_gej);
+    /* We declassify the pubkey sum because serializing a group element (done in the
+     * `_calculate_input_hash_scalar` call following) is not a constant-time operation.
+     */
+    secp256k1_declassify(ctx, &prevouts_pubkey_sum_ge, sizeof(prevouts_pubkey_sum_ge));
+
+    /* Calculate the input_hash and convert it to a scalar so that it can be multiplied with the summed up private keys, i.e., a_sum = a_sum * input_hash.
+     * By multiplying the scalars together first, we can save an elliptic curve multiplication.
+     *
+     * Note: _input_hash_scalar can only fail if the output of the hash function is greater than the curver order, which is statistically improbable.
+     * Returning an error here results in an untestable branch in the code, but we do this anyways to ensure strict compliance with BIP0352.
+     */
+    if (!secp256k1_silentpayments_calculate_input_hash_scalar(&input_hash_scalar, outpoint_smallest36, &prevouts_pubkey_sum_ge)) {
+        return 0;
+    }
```

Comment:
in the newly introduced failure branch of of the sending function: missing clearing of `seckey_sum_scalar` detected :detective:

Reply:
One of these days, I promise: I will learn how to spell statiscaly , and I will remember to clear my scalars 😅